error[E0016]: blocks in constant functions are limited to items and tail expressions
  --> $DIR/const-fn-error.rs:16:19
   |
16 |     let mut sum = 0;
   |                   ^

error[E0015]: calls in constant functions are limited to constant functions, struct and enum constructors
  --> $DIR/const-fn-error.rs:18:5
   |
18 | /     for i in 0..x {
19 | |         //~^ ERROR E0015
20 | |         //~| ERROR E0019
21 | |         sum += i;
22 | |     }
   | |_____^

error[E0019]: constant function contains unimplemented expression type
  --> $DIR/const-fn-error.rs:18:5
   |
18 | /     for i in 0..x {
19 | |         //~^ ERROR E0015
20 | |         //~| ERROR E0019
21 | |         sum += i;
22 | |     }
   | |_____^

error[E0080]: constant evaluation error
  --> $DIR/const-fn-error.rs:28:19
   |
28 |     let a : [i32; f(X)];
   |                   ^^^^ miri failed: machine error: Cannot evaluate within constants: "calling non-const fn `<I as std::iter::IntoIterator><std::ops::Range<usize>>::into_iter`"

error: aborting due to 4 previous errors

